src/xtgeo/grid3d/_gridprops_import_eclrun.py

```diff
@@ -113,37 +138,61 @@ def _import_ecl_output_v2_init(self, pfile, names, grid, strict):
     )
     for result in results:
         prop = GridProperty()
-        self._names.append(result["name"])
+        grid_properties._names.append(result["name"])
         result["name"] = decorate_name(result["name"], grid.dualporo, fracture=False)
         for attr, value in result.items():
             setattr(prop, "_" + attr, value)
 
-        self._props.append(prop)
-        self._dates.append(prop._date)
+        grid_properties._props.append(prop)
+        grid_properties._dates.append(prop._date)
 
-    self._ncol = grid.ncol
-    self._nrow = grid.nrow
-    self._nlay = grid.nlay
+    grid_properties._ncol = grid.ncol
+    grid_properties._nrow = grid.nrow
+    grid_properties._nlay = grid.nlay
 
 
-def _import_ecl_output_v2_rsta(
-    self,
+def import_ecl_restart_gridproperties(
```

Comment:
This function seems to do quite a bit and could benefit from type-hints IMO.

Reply:
Added some typehints.